src/main/scala/com/sksamuel/scapegoat/inspections/unneccesary/UnusedMethodParameter.scala

```diff
@@ -13,34 +13,93 @@ class UnusedMethodParameter extends Inspection {
       import context.global._
       import definitions._
 
-      private def usesParameter(param: ValDef, rhs: Tree): Boolean = {
-        rhs match {
-          case Ident(TermName(name)) => name == param.name.toString
-          case _                     => rhs.children.exists(usesParameter(param, _))
+      private def usesParameter(paramName: String, tree: Tree): Boolean = {
+        tree match {
+          case Ident(TermName(name)) =>
+            name == paramName
+          case _ =>
+            tree.children.exists(usesParameter(paramName, _))
+        }
+      }
+
+      private def usesField(paramName: String, tree: Tree): Boolean = {
+        tree match {
+          case Select(This(_), TermName(name)) =>
+            // FIXME: why is "trim" needed here? Is that a scalac bug?
+            // A test will fail if you take this out!
+            name.trim == paramName
```

Comment:
For some reason, the param name was being reported as "x " by scalac. I couldn't figure out why.